### PR TITLE
[Windows] Allow overwriting the cache's Dart SDK archive license file

### PR DESCRIPTION
**Pull request by loic-sharma**
https://github.com/flutter/engine/pull/43974 added a license file to the Dart SDK's ZIP archive which gets extracted to `flutter/bin/cache/LICENSE.dart_sdk_archive.md`. As a result, extracting the Dart SDK now needs to update the cache's `LICENSE.dart_sdk_archive.md` file.

Windows ZIP extraction does not enable overwriting files. Thus, this change renames the cache's existing Dart SDK license file before extracting the Dart SDK archive.

This is a short-term solution that will be cherry-picked for the next [3.14 beta release](https://github.com/flutter/flutter/issues/132267). Addresses https://github.com/flutter/flutter/issues/132592.

The long-term solution is tracked by https://github.com/flutter/flutter/issues/132702

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Hixie** [comment]
would it be more resilient to future changes if we had a more general way of doing this rather than needing to know the exactly list of files we were expecting?

File: packages/flutter_tools/test/integration.shard/batch_entrypoint_test.dart [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+    // Run the Dart batch entrypoint again to trigger the Dart SDK update.
+    final stdout = await runDartBatch();
+
+    // Check the Dart SDK was re-downloaded and extracted without overwriting
+    // unexpected files.
+    // See: https://github.com/flutter/flutter/issues/132592
+    expect(stdout, contains('Downloading Dart SDK from Flutter engine ...'));
+    expect(stdout, contains('Expanding downloaded archive...'));
+    expect(stdout, isNot(contains('Use the -Force parameter')));
```
Comment (christopherfujino) [inline packages/flutter_tools/test/integration.shard/batch_entrypoint_test.dart:36]: So when I run this test locally without the powershell change from this PR, it hangs indefinitely (as expected). @Hixie is this a reasonable place to set a timeout with an explicit `fail()`, that way if it starts failing, people don't just spam re-run?

Comment (christopherfujino) [reply packages/flutter_tools/test/integration.shard/batch_entrypoint_test.dart:36]: in discord, Hixie had a good idea, to set a Timer for something less than the overall test runner timeout time, which I believe is 12 minutes, and have that timer print out a message saying:

> this is running for much longer than expected and this test is explicitly verifying something doesn't hang, so if it times out that means the test has failed and is unlikely to be a flake.

File: packages/flutter_tools/test/integration.shard/batch_entrypoint_test.dart [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    // Run the Dart entrypoint once to ensure the Dart SDK is downloaded.
+    await runDartBatch();
+
+    // Remove the Dart SDK stamp to cause the Dart batch entrypoint to
+    // re-download the Dart SDK.
+    expect(dartSdkStamp.existsSync(), true);
+    dartSdkStamp.deleteSync();
+
+    // Run the Dart batch entrypoint again to trigger the Dart SDK update.
+    final stdout = await runDartBatch();
```
Comment (christopherfujino) [inline packages/flutter_tools/test/integration.shard/batch_entrypoint_test.dart:29]:
```
   info • Missing type annotation • packages/flutter_tools/test/integration.shard/batch_entrypoint_test.dart:29:5 • always_specify_types
```

File: packages/flutter_tools/test/integration.shard/batch_entrypoint_test.dart [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+import '../src/common.dart';
+import 'test_utils.dart';
+
+final String flutterRootPath = getFlutterRoot();
+final Directory flutterRoot = fileSystem.directory(flutterRootPath);
+
+Future<void> main() async {
+  // Regression test for https://github.com/flutter/flutter/issues/132592
+  test('flutter/bin/dart updates the Dart SDK', () async {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/integration.shard/batch_entrypoint_test.dart:19]: can you add "without hanging" as a further clue to a future framework gardener?

```suggestion
  test('flutter/bin/dart updates the Dart SDK without hanging', () async {
```

Comment (loic-sharma) [reply packages/flutter_tools/test/integration.shard/batch_entrypoint_test.dart]: Great idea, added!

File: packages/flutter_tools/test/integration.shard/batch_entrypoint_test.dart (hunk 2 above)
Comment (loic-sharma) [reply packages/flutter_tools/test/integration.shard/batch_entrypoint_test.dart]: It turns out that developing Dart while also repeatedly clobbering your Dart SDK is hard 😂

File: packages/flutter_tools/test/integration.shard/batch_entrypoint_test.dart (hunk 1 above)
Comment (loic-sharma) [reply packages/flutter_tools/test/integration.shard/batch_entrypoint_test.dart]: Added! :)

**Comment by loic-sharma** [comment]
> would it be more resilient to future changes if we had a more general way of doing this rather than needing to know the exactly list of files we were expecting?

Summary of this [Discord conversation](https://discord.com/channels/608014603317936148/608022056616853515/1141473690681360575): This fix is time sensitive as it unblocks the [3.14 beta release](https://github.com/flutter/flutter/issues/132267). A more robust solution is tracked by https://github.com/flutter/flutter/issues/132702.

**Review by christopherfujino — APPROVED** [review]
LGTM

**Review by yaakovschectman — APPROVED** [review]

**Comment by yaakovschectman** [comment]
This is introducing test failures for the windows, linux, and mac `framework_test_libraries` on the tree build.

**Pull request merged**